src/storage/engine/rocksdb.rs

```diff
@@ -39,15 +40,37 @@ impl EngineRocksdb {
             }
             _ => (path.to_owned(), None),
         };
+        Ok(EngineRocksdb {
+            db: try!(Self::open_or_create_db(&path, cfs)),
+            temp_dir: temp_dir,
+        })
+    }
+
+    fn open_or_create_db(path: &str, cfs: &[CfName]) -> Result<DB> {
+        let mut opts = Options::new();
+        opts.create_if_missing(false);
+        match DB::open_cf(&opts, path, cfs) {
+            Ok(db) => return Ok(db),
+            Err(e) => warn!("open rocksdb fail: {}", e),
```

Comment:
@BusyJay Rocksdb requires to open with all existing column family names. So we first try to open an old db (with all column families we want), then try to create one if step 1 fails.